### PR TITLE
apply propIter to Join

### PR DESCRIPTION
**Pull request by nevermore3**
fetch prop on player 'Tim Duncan' | go from $-.VertexID over like yield like._dst 
Will Error Because JoinIter Don't support PropIter

### CODE REVIEW

**Review by yixinglu — DISMISSED** [review]

**Review by Shylock-Hg — APPROVED** [review]

File: src/executor/query/DataJoinExecutor.cpp [hunk 1]
```diff
@@ -36,7 +36,7 @@ folly::Future<Status> DataJoinExecutor::doInnerJoin() {
                        .iter();
     DCHECK(!!lhsIter);
     VLOG(1) << "lhs: " << dataJoin->leftVar().first << " " << lhsIter->size();
-    if (!lhsIter->isSequentialIter() && !lhsIter->isJoinIter()) {
+    if (lhsIter->isGetNeighborsIter() || lhsIter->isDefaultIter()) {
```
Comment (Shylock-Hg) [inline src/executor/query/DataJoinExecutor.cpp:39]: I prefer to use `switch` statement.

File: src/context/Iterator.cpp [hunk 2]
```diff
@@ -347,34 +347,71 @@ const Value& SequentialIter::getColumn(int32_t index) const {
...
+            }
+            case Iterator::Kind::kJoin: {
+                nextSeg = buildIndexFromJoinIter(static_cast<const JoinIter*>(lhs), 0);
+                break;
+            }
+            case Iterator::Kind::kProp: {
+                nextSeg = buildIndexFromPropIter(static_cast<const PropIter*>(lhs), 0);
+                break;
+            }
+            default: {
```
Comment (Shylock-Hg) [inline src/context/Iterator.cpp:364]: List all value instead of default.

File: src/context/Iterator.cpp [hunk 3]
```diff
@@ -347,34 +347,71 @@ const Value& SequentialIter::getColumn(int32_t index) const {
...
+        }
+        case Iterator::Kind::kJoin: {
+            buildIndexFromJoinIter(static_cast<const JoinIter*>(rhs), nextSeg);
+            break;
+        }
+        case Iterator::Kind::kProp: {
+            buildIndexFromPropIter(static_cast<const PropIter*>(rhs), nextSeg);
+            break;
+        }
+        default: {
```
Comment (Shylock-Hg) [inline src/context/Iterator.cpp:385]: ditto.

File: src/context/Iterator.cpp (hunk 2 above)
Comment (nevermore3) [reply src/context/Iterator.cpp]: done

File: src/context/Iterator.cpp (hunk 3 above)
Comment (nevermore3) [reply src/context/Iterator.cpp]: done

**Review by Shylock-Hg — APPROVED** [review]

**Review by CPWstatic — APPROVED** [review]

**Pull request merged**